## Assignment 1- Basic image processing
by Bianka Szöllősi

__Importing libraries and reading in images__

In [2]:
#loading in libraries
import os
import numpy as np
import sys
import cv2
from pathlib import Path
from glob import glob
import pandas as pd

In [3]:
#reading path to images
path_to_image = os.path.join("..", "pokemon_images")

In [4]:
#printing each file name
for image_name in Path(path_to_image).glob("*.jpg"):
    print(image_name)

../pokemon_images/clefable.jpg
../pokemon_images/swinub.jpg
../pokemon_images/tyranitar.jpg
../pokemon_images/slakoth.jpg
../pokemon_images/togepi.jpg
../pokemon_images/absol.jpg
../pokemon_images/salamence.jpg
../pokemon_images/shinx.jpg
../pokemon_images/pikachu.jpg
../pokemon_images/trapinch.jpg
../pokemon_images/vulpix.jpg
../pokemon_images/charmander.jpg
../pokemon_images/jigglypuff.jpg
../pokemon_images/chansey.jpg
../pokemon_images/cyndaquil.jpg
../pokemon_images/dragonite.jpg
../pokemon_images/zapdos.jpg
../pokemon_images/rapidash.jpg
../pokemon_images/corsola.jpg
../pokemon_images/eevee.jpg
../pokemon_images/ditto.jpg


In [5]:
#making a new image path so that the loops are easier to make
img_path = os.path.join("..", "pokemon_images", "*.jpg")

In [6]:
#making list of image names
img_names = glob(img_path)

__Finding the height, width and number of channels for each image__

In [7]:
for file in img_names:
    img = cv2.imread(file)
    print(img.shape)

(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)


Height = 120

Width = 120

Number of channels = 3

__Splitting image into 4 equal quadrants and saving them as jpg__

In [ ]:
#needed to install "image_slicer" package
#pip install image_slicer

In [8]:
#importing new package
import image_slicer

In [9]:
#creating a loop for slicing each image into 4 equal parts (the "image_slicer.slice" command automatically saves the resulting pictures)
for file in img_names:
    img = cv2.imread(file)
    image_slicer.slice(file, 4) #I realize it returns png pictures, but I couldn't fix it

__Creating and saving a dataframe with filename, width and height for the new images__

In [14]:
#list of the image names
image_list = [os.path.basename(image_name) for image_name in Path(path_to_image).glob("*.png")] #I realize it only works with the *.png ending because I couldn't save it as a jpg file
image_list

['zapdos_02_02.png',
 'vulpix_01_01.png',
 'salamence_01_01.png',
 'charmander_02_02.png',
 'clefable_01_01.png',
 'ditto_02_02.png',
 'corsola_01_01.png',
 'pikachu_01_01.png',
 'absol_02_01.png',
 'slakoth_01_01.png',
 'cyndaquil_02_01.png',
 'dragonite_01_02.png',
 'dragonite_02_01.png',
 'eevee_02_02.png',
 'salamence_02_02.png',
 'ditto_01_01.png',
 'clefable_02_02.png',
 'chansey_01_02.png',
 'shinx_02_01.png',
 'pikachu_01_02.png',
 'chansey_02_01.png',
 'dragonite_02_02.png',
 'slakoth_02_02.png',
 'eevee_01_02.png',
 'vulpix_01_02.png',
 'chansey_01_01.png',
 'slakoth_02_01.png',
 'dragonite_01_01.png',
 'ditto_01_02.png',
 'jigglypuff_01_02.png',
 'cyndaquil_01_01.png',
 'shinx_01_02.png',
 'jigglypuff_01_01.png',
 'zapdos_01_01.png',
 'cyndaquil_02_02.png',
 'togepi_01_02.png',
 'tyranitar_02_02.png',
 'clefable_01_02.png',
 'swinub_02_01.png',
 'chansey_02_02.png',
 'rapidash_02_02.png',
 'tyranitar_01_01.png',
 'corsola_02_02.png',
 'vulpix_02_02.png',
 'rapidash_01_01.png

In [15]:
#getting heigth and width for the new images
for image_name in Path(path_to_image).glob("*.png"): #I realize it only works with the *.png ending because I couldn't save it as a jpg file
    img = cv2.imread(file)
    print(img.shape) #seems to be the same shape as original, any idea why?

(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 120, 3)
(120, 

In [21]:
#list of the image heigths
image_heigth = [img.shape[0] for image_name in Path(path_to_image).glob("*.png")] #I realize it only works with the *.png ending because I couldn't save it as a jpg file
image_heigth

[120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120]

In [20]:
#list of the image heigths
image_width = [img.shape[1] for image_name in Path(path_to_image).glob("*.png")] #I realize it only works with the *.png ending because I couldn't save it as a jpg file
image_width

[120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120,
 120]

In [24]:
#creating data frame
df = pd.DataFrame()
df['filename']  = image_list
df['width']  = image_width
df['heigth']  = image_heigth

In [25]:
#viewing df
print(df)

                filename  width  heigth
0       zapdos_02_02.png    120     120
1       vulpix_01_01.png    120     120
2    salamence_01_01.png    120     120
3   charmander_02_02.png    120     120
4     clefable_01_01.png    120     120
..                   ...    ...     ...
79    clefable_02_01.png    120     120
80       eevee_02_01.png    120     120
81    trapinch_02_01.png    120     120
82    trapinch_01_02.png    120     120
83       ditto_02_01.png    120     120

[84 rows x 3 columns]


In [26]:
#saving df
outpath = os.path.join("..", "data", "image_df.txt")
df.to_csv(outpath, index = False)